src/proto/single/resp.rs

```diff
@@ -117,6 +120,88 @@ pub async fn read_ok<R: AsyncBufReadExt + Unpin>(r: R) -> Result<(), Error> {
     Err(bad("server ok", &rr).into())
 }
 
+// ----------------------------------------------
+
+/// Faktory service stats.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct FaktoryServiceStats {
+    /// Total number of job failures.
+    pub total_failures: u64,
+
+    /// Total number of processed jobs.
+    pub total_processed: u64,
+
+    /// Total number of enqueued jobs.
+    pub total_enqueued: u64,
+
+    /// Total number of queues.
+    pub total_queues: u64,
+
+    /// Queues stats.
+    ///
+    /// A mapping between a queue name and its size (number of jobs on the queue).
+    /// The keys of this map effectively make up a list of queues that are currently
+    /// registered in the Faktory service.
+    pub queues: HashMap<String, u64>,
+
+    /// Faktory's task runner stats.
+    ///
+    /// Note that this is exposed as a "generic" `serde_json::Value` since this info
+    /// belongs to deep implementation details of the Faktory service.
+    pub tasks: serde_json::Value,
```

Comment:
This is kept as `serde_json::Value` since those are super deep implementation details:
```sh
tasks: Object {
            "Busy": Object {
                "reaped": Number(0),
                "size": Number(0),
            },
            "Dead": Object {
                "cycles": Number(0),
                "enqueued": Number(0),
                "size": Number(0),
                "wall_time_sec": Number(0),
            },
            "Retries": Object {
                "cycles": Number(1),
                "enqueued": Number(0),
                "size": Number(0),
                "wall_time_sec": Number(0.000170441),
            },
            "Scheduled": Object {
                "cycles": Number(1),
                "enqueued": Number(0),
                "size": Number(0),
                "wall_time_sec": Number(0.000316549),
            },
            "Workers": Object {
                "reaped": Number(0),
                "size": Number(1),
            },
        },
```

If they need those - they definitely will find their way in via `serde_json::Values::get`.

Reply:
wdyt?

Reply:
The `tasks` field is [commented as deprecated](https://github.com/contribsys/faktory/pull/474/files#diff-4bbbde2a77215c352f9e99f0f2764fb9fb09eaa9052433b58a474b33019f7486R23), and so I guess not only should this stay as generic `serde_json::Value` but also annotated as `#[deprecated]`